contracts/cw20-base/src/contract.rs

```diff
@@ -377,6 +380,32 @@ pub fn execute_send(
     Ok(res)
 }
 
+pub fn execute_update_minter(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    new_minter: String,
+) -> Result<Response, ContractError> {
+    let mut config = TOKEN_INFO.load(deps.storage)?;
+    // Check that sender is authorized to update minter
+    if config.mint.is_none() || config.mint.as_ref().unwrap().minter != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let minter = deps.api.addr_validate(&new_minter)?;
+    let minter_data = MinterData {
+        minter,
+        cap: config.mint.unwrap().cap,
+    };
+    config.mint = Some(minter_data);
+
+    TOKEN_INFO.save(deps.storage, &config)?;
+
+    Ok(Response::default()
+        .add_attribute("action", "update_minter")
+        .add_attribute("minter", new_minter))
```

Comment:
```suggestion
        .add_attribute("new_minter", new_minter))
```
looks more explanatory.